File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prefix = [1]
        for num in nums:
            prefix.append(prefix[-1] * num)
        
        postfix = [1]
        for i in range(len(nums) - 1, -1, -1):
            num = nums[i]
            postfix.append(postfix[-1] * num)
        
        postfix = postfix[::-1]
        result = []
        for i in range(len(nums)):
            result.append(prefix[i] * postfix[i + 1])
        
        return result

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> result;
        result.push_back(1);
        
        for (int num: nums) {
            result.push_back(result.back() * num);
        }
        
        result.pop_back();
        int back = 1;
        for (int i = nums.size() - 1; i >= 0; i--) {
            result[i] *= back;
            back *= nums[i];
        }
        
        return result;
    }
};